### PR TITLE
[WIP] Update to latest snapshot

### PR DESCRIPTION
**Pull request by hasufell**
Depends on https://github.com/input-output-hk/cardano-db-sync/pull/372

### CODE REVIEW

File: src/Cardano/Db/Database.hs [hunk 1]
```diff
@@ -93,7 +94,9 @@ runActions trce env plugin actions = do
             runRollbacks trce plugin pt
             dbAction Continue ys
         (ys, zs) -> do
-          insertBlockList trce env plugin ys
+          insertBlockList trce env
+            undefined -- ??
```
Comment (hasufell) [inline src/Cardano/Db/Database.hs:98]: I don't know what to pass here. This is `LedgerStateVar`, which isn't even exposed, see https://github.com/input-output-hk/cardano-db-sync/issues/373

File: src/Cardano/SmashDbSync.hs [hunk 2]
```diff
@@ -191,8 +190,8 @@ runDbSyncNode plugin enp =
     orDie renderDbSyncNodeError $ do
       liftIO . logInfo trce $ "Reading genesis config."
 
-      genCfg <- readGenesisConfig enc
-      genesisEnv <- hoistEither $ genesisConfigToEnv genCfg
+      genCfg <- readCardanoGenesisConfig enc
+      genesisEnv <- hoistEither $ genesisConfigToEnv undefined genCfg
```
Comment (hasufell) [inline src/Cardano/SmashDbSync.hs:194]: Same here, don't know what to pass. This is `DbSyncNodeParams`.

File: src/Cardano/SmashDbSync.hs [hunk 3]
```diff
@@ -543,7 +541,8 @@ chainSyncClient trce metrics latestPoints currentTip actionQueue =
                           (mNodeHeight metrics)
                 let dummySlotDetails =
                       SlotDetails {
-                        sdTime      = UTCTime (Time.fromGregorian 1970 1 1) 0
+                        sdSlotTime  = UTCTime (Time.fromGregorian 1970 1 1) 0
+                      , sdCurrentTime  = UTCTime (Time.fromGregorian 1970 1 1) 0 -- ??
```
Comment (hasufell) [inline src/Cardano/SmashDbSync.hs:545]: TODO: get current time

**Review by hasufell — COMMENTED** [review]
There are a couple of things I'm not sure about wrt API. It would be good if @erikd could chime in.

**Comment by ksaric** [comment]
This could be useful if you can do this, I tried bumping up the version but ran into issues with several snapshots.

**Comment by hasufell** [comment]
> This could be useful if you can do this, I tried bumping up the version but ran into issues with several snapshots.

It already does build here. But there are those two issues I need assistance with and was hoping @erikd can clear that up.

**Comment by erikd** [comment]
In order to make progress on this, you will need to submit a PR to `db-sync` that exposes the types you need.

**Comment by hasufell** [comment]
@erikd can you comment on the points above?

* https://github.com/input-output-hk/smash/pull/99#discussion_r509391901
* https://github.com/input-output-hk/smash/pull/99#discussion_r509392887

**Pull request closed without merging**